clients/instance/ibm-pi-workspaces.go

```diff
@@ -54,3 +70,58 @@ func (f *IBMPIWorkspacesClient) GetAll() (*models.Workspaces, error) {
 	}
 	return resp.Payload, nil
 }
+
+// Create a workspace
+func (f *IBMPIWorkspacesClient) Create(name, location, groupID, plan string) (*resourcecontrollerv2.ResourceInstance, *core.DetailedResponse, error) {
+	resourceController, err := ibmpisession.CreateResourceControllerV2(f.session.Options.URL, f.session.Options.Authenticator)
+	if err != nil {
+		return nil, nil, fmt.Errorf("error creating Resource Controller client: %v", err)
+	}
+	planID := translatePlan(plan)
+	if planID == "" {
+		return nil, nil, fmt.Errorf("workspace creation error, incorrect plan value; either \"public\" or \"private\" is allowed")
+	}
+	params := resourceController.NewCreateResourceInstanceOptions(name, location, groupID, planID)
+	controller, response, err := resourceController.CreateResourceInstance(params)
+	if err != nil {
+		return nil, response, fmt.Errorf("error creating workspace: controller %v response %v  err %v", controller, response, err)
+	}
+	if response.StatusCode >= 400 {
+		return nil, response, fmt.Errorf("error creating resource instance. Status code: %d", response.StatusCode)
+	}
+	return controller, response, nil
+}
+
+// Delete a workspace
+func (f *IBMPIWorkspacesClient) Delete(workspaceID string) (*core.DetailedResponse, error) {
+	resourceController, err := ibmpisession.CreateResourceControllerV2(f.session.Options.URL, f.session.Options.Authenticator)
+	if err != nil {
+		return nil, fmt.Errorf("error creating Resource Controller client: %v", err)
+	}
+	params := resourceController.NewDeleteResourceInstanceOptions(workspaceID)
+	response, err := resourceController.DeleteResourceInstance(params)
+	if err != nil {
+		return response, fmt.Errorf("error deleting workspace: response %v, err %v", response, err)
+	}
+	if response.StatusCode >= 400 {
+		return response, fmt.Errorf("error deleting resource instance. Status code: %d", response.StatusCode)
+	}
+	return response, nil
+}
+
+// Get a resourceController
+func (f *IBMPIWorkspacesClient) GetRC(rcWorkspaceID string) (*resourcecontrollerv2.ResourceInstance, *core.DetailedResponse, error) {
```

Comment:
```suggestion
// Get a workspace
func (f *IBMPIWorkspacesClient) Get(workspaceID string) (*resourcecontrollerv2.ResourceInstance, *core.DetailedResponse, error) {
```

Reply:
Why different `Get` approaches? Do they give different data with RC?

Reply:
Yes the Get is the powervs workspace get endpoint, and the Get resource controller is the resource controller's get endpoint. They return slightly different data, and the our get endpoint is only implement in powervs not stratos.

Reply:
Do we need both versions?